doc: manual: constructors: note the correspondence with callable objs (#53051)
JuliaLang/julia: doc/src/manual/constructors.md

specifying parameters for the type to be constructed, i.e. this call will return a `SummedArray{T,S}`.
`new{T,S}` can be used in any constructor definition, but for convenience the parameters
to `new{}` are automatically derived from the type being constructed when possible.

> ## [Constructors are just callable types](@id man-constructors-callable-types)

Reviewer: ```suggestion
## Constructors are just callable objects
```
nit: better to let most id be auto-generated

An object of any type may be [made callable](@ref "Function-like objects") by defining a
method. This includes types, i.e., objects of type [`Type`](@ref); and constructors may,

> in fact, be viewed as just callable type objects. Example:
> 
> ```jldoctest
> julia> struct S
>            f::Int
>        end
> 
> julia> struct T end
> 
> julia> S(::T) = S(7)
> S
> 
> julia> S(T())
> S(7)
> 
> julia> (::Type{S})(::T) = S(8)  # overwrites the above constructor
> 
> julia> S(T())
> S(8)
> ```

Reviewer: I feel like this example is not particularly clear to me, since it just creates a lot of new types which don't do anything. Perhaps instead show an existing function that does have a lot of variety in methods defined for it?

````suggestion
in fact, be viewed as just callable type objects.
For example, there are many methods defined on `Int` and various supertypes of it:

```julia-repl
julia> methods(Int)
# 14 methods for type constructor:
  [1] Int64(x::Float64)
     @ Base float.jl:959
  [2] Int64(x::Float32)
     @ Base float.jl:959
  [3] Int64(x::Float16)
     @ Base float.jl:959
  [4] Int64(x::Ptr)
     @ boot.jl:896
  [5] Int64(x::Union{Bool, Int32, Int64, UInt16, UInt32, UInt64, UInt8, Int128, Int16, Int8, UInt128})
     @ boot.jl:886
  [6] (dt::Type{<:Integer})(ip::Sockets.IPAddr)
     @ Sockets ~/julia1/usr/share/julia/stdlib/v1.11/Sockets/src/IPAddr.jl:11
  [7] (::Type{T})(x::BigFloat) where T<:Integer
     @ Base.MPFR mpfr.jl:403
  [8] (::Type{T})(x::Enum{T2}) where {T<:Integer, T2<:Integer}
     @ Base.Enums Enums.jl:19
  [9] (::Type{T})(x::Rational) where T<:Integer
     @ Base rational.jl:140
 [10] (::Type{T})(z::Complex) where T<:Real
     @ Base complex.jl:44
 [11] (::Type{T})(x::Base.TwicePrecision) where T<:Number
     @ Base twiceprecision.jl:265
 [12] (::Type{T})(x::T) where T<:Number
     @ boot.jl:894
 [13] (::Type{T})(x::BigInt) where T<:Union{Int128, Int16, Int32, Int64, Int8}
     @ gmp.jl:373
 [14] (::Type{T})(x::AbstractChar) where T<:Union{Int32, Int64}
     @ char.jl:51
````

Author: The motivation for my example was to show that the syntax `(::Type{S})(::T) = ...` is exactly equivalent to `S(::T) = ...`, to the degree that the methods overwrite each other. I don't think that's made clear in any other place in the docs. So I'd like to have both examples if possible.

I chose `Bool` instead of `Int` for the `methods` example, because it doesn't have quite so many methods as `Int` does.

defined on `Bool` and various supertypes of it:

```julia-repl
julia> methods(Bool)
# 10 methods for type constructor:
  [1] Bool(x::BigFloat)
     @ Base.MPFR mpfr.jl:393
  [2] Bool(x::Float16)
     @ Base float.jl:338
  [3] Bool(x::Rational)
     @ Base rational.jl:138
  [4] Bool(x::Real)
     @ Base float.jl:233
  [5] (dt::Type{<:Integer})(ip::Sockets.IPAddr)
     @ Sockets ~/tmp/jl/jl/julia-nightly-assert/share/julia/stdlib/v1.11/Sockets/src/IPAddr.jl:11
  [6] (::Type{T})(x::Enum{T2}) where {T<:Integer, T2<:Integer}
     @ Base.Enums Enums.jl:19
  [7] (::Type{T})(z::Complex) where T<:Real
     @ Base complex.jl:44
  [8] (::Type{T})(x::Base.TwicePrecision) where T<:Number
     @ Base twiceprecision.jl:265
  [9] (::Type{T})(x::T) where T<:Number
     @ boot.jl:894
 [10] (::Type{T})(x::AbstractChar) where T<:Union{AbstractChar, Number}
     @ char.jl:50
```

The usual constructor syntax is exactly equivalent to the function-like object
syntax, so trying to define a method with each syntax will cause the first method
to be overwritten by the next one:

```jldoctest
julia> struct S
           f::Int
       end

> julia> struct T end
> 
> julia> S(::T) = S(7)
> S
> 
> julia> S(T())
> S(7)
> 
> julia> (::Type{S})(::T) = S(8)  # overwrites the above constructor
> 
> julia> S(T())

Reviewer: suggestion: just get rid of `struct T`, since it doesn't seem to contribute anything to the example:
```suggestion

julia> S() = S(7)
S

julia> S()
S(7)

julia> (::Type{S})() = S(8)  # overwrites the above constructor

julia> S()
```
